src/libsync/creds/httpcredentials.cpp

```diff
@@ -259,13 +249,25 @@ bool HttpCredentials::refreshAccessToken()
     _ready = false;
 
     OAuth *oauth = new OAuth(_account, this);
+    connect(oauth, &OAuth::refreshError, this, [oauth, this](QNetworkReply::NetworkError error, const QString &) {
+        oauth->deleteLater();
+        _isRenewingOAuthToken = false;
+        if (error == QNetworkReply::AuthenticationRequiredError) {
```

Comment:
WDYM?

Reply:
A http error that indicates we should better start fresh

Reply:
Might have to look into the oC server code to give a definite answer. But it might be better to just check every possible error code.

I think the following list covers most problems where retrying makes sense:

```
TemporaryNetworkFailureError
NetworkSessionFailedError
InternalServerError
ServiceUnavailableError
UnknownNetworkError
```

I skipped the following ones because they indicate potential attacks or at least serious misconfigurations:

```
SslHandshakeFailedError
TooManyRedirectsError
InsecureRedirectError
ContentNotFoundError
UnknownContentError
ProtocolFailure
UnknownServerError
```

These make no sense in this context and should never be returned by the server:

```
ContentReSendError
ContentConflictError
ContentGoneError
OperationNotImplementedError
```

The other proxy related ones imply a misconfiguration that we can't fix, thus retrying the request will most likely lead to the same error again.